legend-sdlc-server/src/main/java/org/finos/legend/sdlc/server/gitlab/api/GitLabPatchApi.java

```diff
@@ -0,0 +1,357 @@
+        {
+            throw new LegendSDLCServerException(patchReleaseVersionId.toVersionIdString() + " already exists", Response.Status.BAD_REQUEST);
+        }
+
+        // Check if the patch branch we want to release exists or not
+        Branch patchBranch = null;
+        try
+        {
+            patchBranch = getGitLabApi().getRepositoryApi().getBranch(gitLabProjectId.getGitLabId(), getPatchReleaseBranchName(patchReleaseVersionId));
+        }
+        catch (Exception e)
+        {
+            LOGGER.warn("Error in fetching the patch release branch ", patchReleaseVersionId, projectId, e);
+        }
+        if (patchBranch == null)
+        {
+            throw new LegendSDLCServerException("Patch release branch for " + patchReleaseVersionId + " doesn not exist for the project " + projectId, Response.Status.BAD_REQUEST);
+        }
+        Version releaseVersion =  newVersion(gitLabProjectId, patchReleaseVersionId, patchBranch.getCommit().getId(), VersionId.newVersionId(patchReleaseVersionId.getMajorVersion(), patchReleaseVersionId.getMinorVersion(), patchReleaseVersionId.getPatchVersion()), "");
+
+        // delete the patch release branch and close the MRs created for this branch
+        submitBackgroundRetryableTask(() -> GitLabApiTools.deleteBranchAndVerify(getGitLabApi().getRepositoryApi(), gitLabProjectId.getGitLabId(), getPatchReleaseBranchName(patchReleaseVersionId), 20, 1_000), 5000L, "delete " + getPatchReleaseBranchName(patchReleaseVersionId));
+        submitBackgroundRetryableTask(() -> closeMergeRequestsCreatedForPatchReleaseBranch(gitLabProjectId, patchReleaseVersionId), 5000L, "close merge requests created for branch " + getPatchReleaseBranchName(patchReleaseVersionId));
+        return releaseVersion;
+    }
+
+    private boolean closeMergeRequestsCreatedForPatchReleaseBranch(GitLabProjectId projectId, VersionId patchReleaseVersionId)
+    {
+        List<MergeRequest> mergeRequests = Collections.emptyList();
+        String branchName = getSourceBranch(projectId, patchReleaseVersionId);
+        try
+        {
+           mergeRequests = getGitLabApi().getMergeRequestApi().getMergeRequests(projectId.getGitLabId()).stream().filter(mr -> branchName.equals(mr.getTargetBranch())).collect(Collectors.toList());
+        }
+        catch (Exception e)
+        {
+            LOGGER.warn("Unable to fetch merge requests for the project " + projectId.getGitLabId(), e);
+        }
+        for (MergeRequest mergeRequest : mergeRequests)
+        {
+            try
+            {
+                getGitLabApi().getMergeRequestApi().updateMergeRequest(projectId.getGitLabId(), mergeRequest.getIid(), branchName, null, null, null, Constants.StateEvent.CLOSE, null, null, null, null, null, null);
+            }
+            catch (Exception e)
+            {
+                LOGGER.warn("Unable to close merge request " + mergeRequest.getId() + " of the project " + projectId.getGitLabId(), e);
+            }
+        }
```

Comment:
If there's a failure to close any of them that suggests a retry might work, then the method should return false.